### ISSUE TITLE
TChannel.advertise should be safe to call multiple times

### ISSUE THREAD

**Issue by abhinav** [uber/tchannel-python#319]
Currently, advertise for the sync and async clients do not keep track whether they're already advertising. This means that calling advertise multiple times is unsafe because that'll just start up multiple advertise loops. We should keep track of advertisement status so that it's safe to all advertise multiple times.

**Comment by breerly** [comment]
Not totally sure if we need this. If you really want idempotence, couldn't you just:

```python
from tchannel.sync.singleton import TChannel

if not TChannel.prepared:
    TChannel.prepare(...)
    TChannel.get_instance().advertise(...)
```

and then in other parts of the application you can just:

```python
from tchannel.singleton import TChannel

tchannel = TChannel.get_instance()
```

**Comment by breerly** [comment]
Is there value to making advertise idempotent outside of the singleton facility?

**Comment by abhinav** [comment]
Resolved by #319
